homeassistant/components/loqed/coordinator.py

```diff
@@ -115,22 +115,22 @@ async def ensure_webhooks(self) -> None:
             self.hass, DOMAIN, "Loqed", webhook_id, self._handle_webhook
         )
 
-        if cloud.async_active_subscription(self.hass):
-            webhook_url = await async_cloudhook_generate_url(self.hass, self._entry)
-        else:
-            webhook_url = webhook.async_generate_url(
-                self.hass, self._entry.data[CONF_WEBHOOK_ID]
-            )
-
-        _LOGGER.debug("Webhook URL: %s", webhook_url)
-
         webhooks = await self.lock.getWebhooks()
 
         webhook_index = next(
-            (x["id"] for x in webhooks if x["url"] == webhook_url), None
+            (x["id"] for x in webhooks if webhook_id in x["url"]), None
```

Comment:
This is intended change.
Current setup doesn't work  in some situations - if generated URL has incorrect hostname, or uses Cloudhook service, it will keep recreating incorrect webhook in the gateway, even if corrected manually.

Reply:
In that case I think it's better to provide the user with an option the configure the host part for the webhook instead of doing only a partial comparison as that has the risk of breaking updates for the user if the host of home assistant changes.

Reply:
@mikewoudenberg 
> risk of breaking updates for the user if the host of home assistant changes.
That would also mean polluting webhook list with unnecessary entries.
Shall we also add logic to remove other entries with same `webhook_id`, but different URLs?

Reply:
I'm no expert in the inner workings of hass, but I think this flow is the way to go for the desired feature: https://developers.home-assistant.io/docs/config_entries_options_flow_handler . You could add an update_option listener to clean up old webhooks when the option changes

Reply:
Change could be external (like a change of home network setup). 

Also it doesn't solve issue when generated URL is incorrect.

Reply:
This solution also doesn't as it will not delete a webhook that has a correct ID and incorrect host part. I suggest the following solution:
- add an option flow handler where you can set a base URL for the webhook, default it to the URL that Home Assistant generates (as that is working in most of the cases)
- When the user decides to change the base URL for the webhook via the options flow, delete any webhooks from the bridge that have the same Webhook ID and register a new webhook with the given URL. 

This way the webhook can be fully controlled from Home Assistant (for which it is registered in the first place).